actions/docs-verifier/src/ActionRunner/Program.cs

```diff
@@ -20,9 +19,28 @@
     throw new InvalidOperationException($"The value of GITHUB_PR_NUMBER environment variable is not valid.");
 }
 
+static bool IsYmlOrMarkdownFile(PullRequestFile file) => Path.GetExtension(file.FileName) is ".yml" or ".md";
+
+static bool IsInWhatsNewDirectory(PullRequestFile file)
+{
+    string? whatsNewPath = WhatsNewConfigurationReader.GetWhatsNewPath();
+    if (whatsNewPath is { Length: > 0 })
+    {
+        // Example:
+        // file.FileName:   docs/whats-new/2021-03.md
+        // whatsNewPath:    docs/whats-new
+
+        return whatsNewPath.StartsWith(file.FileName, StringComparison.OrdinalIgnoreCase);
+    }
+
+    return false;
+}
+
 List<PullRequestFile> files = (await GitHubPullRequest.GetPullRequestFilesAsync(pullRequestNumber)).ToList();
 
-foreach (PullRequestFile file in files)
+// We should only ever fail on MD and YML files, no other files require redirection.
+// Also, filter out files that are part of the "What's new" directory - as they shouldn't require redirects.
+foreach (PullRequestFile file in files.Where(f => IsYmlOrMarkdownFile(f) && !IsInWhatsNewDirectory(f)))
```

Comment:
nit: Consider doing ` && (f.IsRenamed() || f.IsRemoved())` before checking whats new. `IsRenamed` and `IsRemoved` are very cheap compared to reading from disk and parsing json.